executor/analyze.go

```diff
@@ -256,7 +256,8 @@ func CollectSamplesAndEstimateNDVs(ctx context.Context, e ast.RecordSet, numCols
 			row.Data = row.Data[1:]
 		}
 		for i, val := range row.Data {
-			err = collectors[i].collect(val)
+			tmp := types.CopyDatum(val)
+			err = collectors[i].collect(tmp)
```

Comment:
How about move the `CopyDatum` in `collect` method? 
Because in `collect` method, we can clearly see that the datum is appended to the `SampleCollector`.
And please add comment about why we need to copy the datum even though we never modify it.